telemetry/run.py

```diff
@@ -0,0 +1,99 @@
+import subprocess
+import threading
+import argparse
+import datetime
+import sys
+import os
+
+
+def parse_cpu_stat(info:list) -> dict:
+    '''
+    Takes data returned by sensors, which contains temperature / voltage data.
+    Returns a dict, whose keys are (group) name of the sensors and values are pairs of (label x data).
+    '''
+    current_group = 'ungrouped'
+    result = dict({ current_group : [] })
+    for line in info:
+        # omit empty lines and `N/A` data
+        if line and not 'N/A' in line:
+            # Adapter names does not contain spaces
+            if ' ' not in line:
+                current_group = line
+                result.update({ line : [] })
+            else:
+                label, data = ' '.join(line.split()).split()[:2]
+                result[current_group].append((label.replace(':', ''), ''.join(data)))
+    # return with empty entries removed
+    return dict(filter(lambda x: len(x[1]), result.items()))
+
+def parse_gpu_stat(info:list) -> list:
+    '''
+    Takes data returned by `nvidia-smi`, which contains a string with two lines, returns
+    the last line, which is the data we need to monitor during an experiment.
+    '''
+    try:
+        # ensure the list is not empty
+        assert info
+    except:
```

Comment:
I don't think you should wrap an assert in a try/except, since an assert is supposed to signify an error that should never happen (unless your assumptions are wrong)

Reply:
Sure. I will change these lines.